hnn_core/extracellular.py

```diff
@@ -0,0 +1,601 @@
+            :meth:`~matplotlib.axes.Axes.set_yticklabels`.
+        show : bool
+            If True, show the figure
+
+        Returns
+        -------
+        fig : instance of plt.fig
+            The matplotlib figure handle into which time series were plotted.
+        """
+        from .viz import plot_extracellular
+
+        if trial_no is None:
+            plot_data = self.get_data()
+        elif isinstance(trial_no, (list, tuple, int, slice)):
+            plot_data = self.get_data()[trial_no, ]
+        else:
+            raise ValueError(f'unkown trial number type, got {trial_no}')
+
+        if isinstance(contact_no, (list, tuple, int, slice)):
+            plot_data = plot_data[:, contact_no, ]
+        elif contact_no is not None:
+            raise ValueError(f'unkown contact number type, got {contact_no}')
+
+        for trial_data in plot_data:
+            fig = plot_extracellular(
+                self.times, trial_data, tmin=tmin, tmax=tmax, ax=ax,
+                decim=decim, color=color,
+                voltage_offset=voltage_offset,
+                voltage_scalebar=voltage_scalebar,
+                contact_labels=contact_labels,
+                show=show)
+        return fig
+
+
+class ExtracellularArrayBuilder(object):
+    """The ExtracellularArrayBuilder class
+
+    Parameters
+    ----------
+    array : ExtracellularArray object
+        The instance of :class:`hnn_core.extracellular.ExtracellularArray` to
+        build in NEURON-Python
+    """
+    def __init__(self, array):
+        self.array = array
+        self.n_contacts = array.n_contacts
+        self._nrn_imem_ptrvec = None
+        self._nrn_imem_vec = None
+        self._nrn_r_transfer = None
+        self._nrn_times = None
+        self._nrn_voltages = None
+
+    def _build(self, cvode=None):
+        """Assemble NEURON objects for calculating extracellular potentials.
+
+        The handler is set up to maintain a vector of membrane currents at at
+        every inner segment of every section of every cell on each CVODE
+        integration step. In addition, it records a time vector of sample
+        times.
+
+        Parameters
+        ----------
+        cvode : instance of h.CVode
+            Multi order variable time step integration method.
+        """
+        # ordered list of h.Sections on this rank (if running in parallel)
+        secs_on_rank = _get_sections_on_this_rank()
+        # np.array of number of segments for each section, ordered as above
+        segment_counts = np.array(_get_segment_counts(secs_on_rank))
+        n_total_segments = segment_counts.sum()
+
+        # pointers assigned to _ref_i_membrane_ at each EACH internal segment
+        self._nrn_imem_ptrvec = h.PtrVector(n_total_segments)
+        # placeholder into which pointer values are read on each sim time step
+        self._nrn_imem_vec = h.Vector(n_total_segments)
+
+        ptr_count = 0
```

Comment:
```suggestion
        ptr_idx = 0
```

I think it's just better to use `xxxx_idx` to refer to indexes and `n_xxxx` to total quantities. Using `xxx_counts` or `counts_xxx` interchangeably confuses me as I don't know if it's a total quantity or an index. E.g., above `segment_counts` is a total quantity